pkg/stores/partition/listprocessor/processor.go

```diff
@@ -189,24 +192,23 @@ func ParseQuery(apiOp *types.APIRequest) *ListOptions {
 	sortOpts := Sort{}
 	sortKeys := q.Get(sortParam)
 	if sortKeys != "" {
-		sortParts := strings.SplitN(sortKeys, ",", 2)
-		primaryField := sortParts[0]
-		if primaryField != "" && primaryField[0] == '-' {
-			sortOpts.primaryOrder = DESC
-			primaryField = primaryField[1:]
-		}
-		if primaryField != "" {
-			sortOpts.primaryField = strings.Split(primaryField, ".")
-		}
-		if len(sortParts) > 1 {
-			secondaryField := sortParts[1]
-			if secondaryField != "" && secondaryField[0] == '-' {
-				sortOpts.secondaryOrder = DESC
-				secondaryField = secondaryField[1:]
+		sortParts := strings.Split(sortKeys, ",")
+		for _, field := range sortParts {
+			sortOrder := ASC
+			if len(field) > 0 {
+				if field[0] == '-' {
+					sortOrder = DESC
+					field = field[1:]
+				}
 			}
-			if secondaryField != "" {
-				sortOpts.secondaryField = strings.Split(secondaryField, ".")
+			if len(field) == 0 {
+				// Current semantics: if we have a sort query like
+				// sort=,metadata.someOtherField
+				// we *DON'T sort
```

Comment:
I suggest we take the chance to fix this - either raising an error or producing reliably the same sorting would be OK FMPOV.

Reply:
The good news is when I made this change a unit test failed.